sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -639,45 +667,14 @@ public String getShareSnapshotId() {
-        List<FileRef> fileRefs = convertResponseAndGetNumOfResults(response);
-
-        if (response.value().nextMarker() == null) {
-            return Flux.fromIterable(fileRefs);
-        }
-        Mono<DirectorysListFilesAndDirectoriesSegmentResponse> listResponse = azureFileStorageClient.directorys().listFilesAndDirectoriesSegmentWithRestResponseAsync(shareName, directoryPath, prefix, snapshot, response.value().nextMarker(), maxResult, null, Context.NONE);
-        Flux<FileRef> fileRefPublisher = listResponse.flatMapMany(newResponse -> nextPageForFileAndDirectories(newResponse, prefix, maxResult));
-        return Flux.fromIterable(fileRefs).concatWith(fileRefPublisher);
-    }
-
-    private Flux<HandleItem> nextPageForHandles(DirectorysListHandlesResponse response, Integer maxResult, boolean recursive) {
-        List<HandleItem> handleItems = response.value().handleList();
-
-        if (response.value().nextMarker() == null) {
-            return Flux.fromIterable(handleItems);
-        }
-        Mono<DirectorysListHandlesResponse> listResponse = azureFileStorageClient.directorys().listHandlesWithRestResponseAsync(shareName, directoryPath, response.value().nextMarker(), maxResult, null, snapshot,  recursive, Context.NONE);
-        Flux<HandleItem> fileRefPublisher = listResponse.flatMapMany(newResponse -> nextPageForHandles(newResponse, maxResult, recursive));
-        return Flux.fromIterable(handleItems).concatWith(fileRefPublisher);
-    }
-
-    private Flux<Integer> nextPageForForceCloseHandles(DirectorysForceCloseHandlesResponse response, String handleId, boolean recursive) {
-        List<Integer> handleCount = Collections.singletonList(response.deserializedHeaders().numberOfHandlesClosed());
-
-        if (response.deserializedHeaders().marker() == null) {
-            return Flux.fromIterable(handleCount);
-        }
-        Mono<DirectorysForceCloseHandlesResponse> listResponse = azureFileStorageClient.directorys().forceCloseHandlesWithRestResponseAsync(shareName, directoryPath, handleId, null, response.deserializedHeaders().marker(), snapshot, recursive, Context.NONE);
-        Flux<Integer> fileRefPublisher = listResponse.flatMapMany(newResponse -> nextPageForForceCloseHandles(newResponse, handleId, recursive));
-        return Flux.fromIterable(handleCount).concatWith(fileRefPublisher);
-    }
-
     private List<FileRef> convertResponseAndGetNumOfResults(DirectorysListFilesAndDirectoriesSegmentResponse response) {
         List<FileRef> fileRefs = new ArrayList<>();
         if (response.value().segment() != null) {
             response.value().segment().directoryItems().forEach(directoryItem -> fileRefs.add(new FileRef(directoryItem.name(), true, null)));
             response.value().segment().fileItems().forEach(fileItem -> fileRefs.add(new FileRef(fileItem.name(), false, fileItem.properties())));
         }
+
+        fileRefs.sort(Comparator.comparing(FileRef::name));
```

Comment:
Why are we adding this sort? This is already a segment of a paged response, what's the point of spending cycles sorting every 1000 file subset of a larger list?

Reply:
The REST spec states that the response is lexically sorted with directories and files intermingled and I wanted to align on this. https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files#remarks

Reply:
If the rest spec says it's sorted, wouldn't it already be sorted?

Reply:
They should sort in our return type in fileRef.
The line here is trying to sort out based on name instead of sort separately based on dir and file.